### PR TITLE
Feature -> Develop | Upgrade Flutter to 3.24.3 and Project Dependencies

### PR DESCRIPTION
**Pull request by we-chintan-p001**
#### [Default Task ID]()
**Time invested:** hh:mm format | **Billable**: 00:00 | **Non-Billable**: 04:00

**Problem Statement**
Fail to run VaahFlutter with new dependencies and new dart or flutter versions

**Describe The Fix/ Solution You Implemented**
Upgraded Flutter & Dependencies

**Proof of your testing (Demo link or video links or image links)**

- For Android
    - https://youtu.be/7du1C7MygU4

- For iOS
    - https://youtu.be/r5pWw1UW_uU

**Dependencies**
- Updated Most of them

### Feature Merge Request Checklist

- [x] I have performed a self-review of my code
- [x] My code does follow `industry standards`
- [x] My code doesn't produce `warnings/ errors` for `dart analyzer`
- [x] New and existing `tests` pass locally with my changes (No tests are there as of now)
- [ ] The code modified as part of this PR has been covered with tests
- [x] My code is `properly formatted`
- [x] I've followed the proper naming conventions
- [x] I have rebased the `feature` on the latest `develop`
- [x] I have run `flutter run` after the rebase
- [x] I have added/updated new dependencies
- [ ] I have read all the `comments` & `notes` in `wireframe` and verified that I did take care of that
- [ ] verify that the `UI` must match with `design` & `wireframe` if available
- [ ] I have updated the `version/ build` (`x.x.x+xxx`)
- [x] I've verified that the latest commit of `develop` exists in your `feature` branch after rebase.
- [x] My fix achieves 100% of what is required
- [ ] Can enhance the solution in the future because better implementation could be made

### CODE REVIEW

File: lib/vaahextendflutter/base/base_controller.dart [hunk 1]
```diff
@@ -9,9 +9,9 @@ import 'package:sentry_flutter/sentry_flutter.dart';
 import '../app_theme.dart';
 import '../env/env.dart';
 import '../services/api.dart';
-import '../services/dynamic_links.dart';
 import '../services/notification/internal/notification.dart';
 import '../services/notification/push/notification.dart';
+import 'root_assets_controller.dart';
 
 class BaseController extends GetxController {
```
Comment (we-khushal-y001) [inline lib/vaahextendflutter/base/base_controller.dart:16]: We can do initialization something like this, just a suggestion

```dart
class AppConfigs {
  AppConfigs({
    required this.configs,
  });

  final Env env;
}

-------------------------------------------------------------------

abstract class LaunchTask {
  const LaunchTask();

  Future<void> initialize(AppConfigs configs);
  Future<void> dispose();
}

class AppLauncher {
  AppLauncher({
    required this.configs,
  });

  final AppConfigs configs;
  final List<LaunchTask> tasks = [];

  void addTask(LaunchTask task) {
    tasks.add(task);
  }

  void addTasks(Iterable<LaunchTask> tasks) {
    tasks.addAll(tasks);
  }

  Future<void> launch() async {
    for (final task in tasks) {
      await task.initialize(configs);
    }
  }

  Future<void> dispose() async {
    for (final task in tasks) {
      await task.dispose();
    }
    tasks.clear();
  }
}

-------------------------------------------------------------------

class BaseController extends GetxController{

  void init(AppConfigs configs)
    final launcher = AppLauncher(configs: configs);
    launcher.addTasks(
     const AlgoliTask();
     .
     .
     .
     // other services
    );
    
     await launcher.launch(); // initialize the tasks
  }
}


-------------------------------------------------------------------

class AlgoliTask extends LaunchTask {
  const AlgoliTask();

  @override
  Future<void> initialize(AppConfigs configs) async {
  // initialize algolia service and add the object into di or singleton class
  // this way other services can utilize this object from  di or singleton class
  }

  @override
  Future<void> dispose() async {}
}
```

Comment (we-chintan-p001) [reply lib/vaahextendflutter/base/base_controller.dart:16]: Good proposal, can we have a separate PR for this one or can you create a separate commit on my branch?

Comment (we-chintan-p001) [reply lib/vaahextendflutter/base/base_controller.dart:16]: Also will we have to do some testing around this?

Comment (we-khushal-y001) [reply lib/vaahextendflutter/base/base_controller.dart:16]: sure

Comment (we-khushal-y001) [reply lib/vaahextendflutter/base/base_controller.dart:16]: Let's go with this approach for now. I'll create new PR for my suggestion.

**Review by we-khushal-y001 — APPROVED** [review]

**Pull request merged**